service/controller/legacy/v25/adapter/guest_iam_policies.go

```diff
@@ -30,7 +30,11 @@ func (i *GuestIAMPoliciesAdapter) Adapt(cfg Config) error {
 	i.WorkerProfileName = key.InstanceProfileName(cfg.CustomObject, key.KindWorker)
 	i.WorkerRoleName = key.RoleName(cfg.CustomObject, key.KindWorker)
 	i.RegionARN = key.RegionARN(cfg.CustomObject)
-	i.KMSKeyARN = cfg.TenantClusterKMSKeyARN
+
+	if cfg.TenantClusterKMSKeyARN != "" {
+		i.KMSKeyARN = cfg.TenantClusterKMSKeyARN
+	}
```

Comment:
When you check the templates you see that there is already a condition to cover this. So IMO this change is not necessary.

Reply:
Yeah I was thinking the same. It's already handled there so it should be enough to have it in the CF template